rust/src/operations/transaction/state.rs

```diff
@@ -43,8 +43,14 @@ impl DeltaTableState {
                     .map(|f| {
                         let field = ArrowField::try_from(f)?;
                         let corrected = match field.data_type() {
-                            // Dictionary encoding columns bools is a waste of space
-                            DataType::Boolean => field.data_type().clone(),
+                            // Dictionary encoding for types around the size of dictionary values does not make much sense.
+                            DataType::Boolean
+                            | DataType::Int8
+                            | DataType::Int16
+                            | DataType::Int32
+                            | DataType::Utf8
```

Comment:
I think you mean `Uint8` here? but honestly I think it still is probably worth it for all the types except boolean. Reasoning explained here: https://github.com/apache/arrow-datafusion/pull/5545#issuecomment-1526917997

Reply:
Indeed I did - glad you caught that! Removed the other values again, as per your comment - makes a lot of sense!